File: 2021/day2/part1.exs
sample_input = "forward 5
down 5
forward 8
up 3
down 8
forward 2" |> String.split("\n")

sample_input
  |> Enum.reduce([0, 0], fn x, [position, depth] -> 
    case x do
      "forward " <> y ->
        [position + String.to_integer(y), depth]
      "down " <> y ->
        [position, depth + String.to_integer(y)]
      "up " <> y ->
        [position, depth - String.to_integer(y)]
    end
  end) 
  |> Enum.product
  |> IO.puts

File: 2021/day2/part1.py
sample_input = """forward 5
down 5
forward 8
up 3
down 8
forward 2""".split("\n")

position = 0
depth = 0
for line in sample_input:
    direction, val = line.split()
    if direction == "forward":
        position += int(val)
    elif direction == "down":
        depth += int(val)
    elif direction == "up":
        depth -= int(val)

print(position * depth)
